cypress/integration/models/administrator/custom-migration-targets/custom-migration-target.ts

```diff
@@ -1,28 +1,59 @@
-import { clickByText, inputText, selectUserPerspective, uploadXml } from "../../../../utils/utils";
+import {
+    click,
+    clickByText,
+    inputText,
+    selectUserPerspective,
+    uploadXml,
+} from "../../../../utils/utils";
 import {
     createNewButton,
     customMigrationTargets,
     button,
     SEC,
     deleteAction,
     editAction,
+    RepositoryType,
 } from "../../../types/constants";
 import { navMenu } from "../../../views/menu.view";
 import { CustomMigrationTargetView } from "../../../views/custom-migration-target.view";
+import { CredentialsSourceControl } from "../../../models/administrator/credentials/credentialsSourceControl";
+
+export enum CustomMigrationTargetOriginType {
+    Repository = "Repository",
+    Manual = "Manual",
+}
+
+export type CMTRepositoryOrigin = {
+    type: CustomMigrationTargetOriginType.Repository;
+    repositoryType: RepositoryType;
+    repositoryUrl: string;
+    branch?: string;
+    rootPath?: string;
+    credentials?: CredentialsSourceControl;
+};
+
+export type CMTManualOrigin = {
+    type: CustomMigrationTargetOriginType.Manual;
+    imagePath?: string;
+    rulesetPath: string;
+};
 
 export interface CustomMigrationTarget {
     name: string;
     description?: string;
-    imagePath?: string;
-    rulesetPath: string;
+    rulesOrigin: CMTRepositoryOrigin | CMTManualOrigin;
```

Comment:
Let's make this more connected to UI forms .

Example : 

ruleType: Repository | Manual

Reply:
Solved in 2e54c55a1fa91b17b4fc06714ac0ca5e489f32b9.